### ISSUE TITLE
Cluster

### ISSUE THREAD

**Issue by Ipmanchess** [tthsqe12/asm#22]
It's not a issue..but more a request and a new challenge ;)

Now that asmFish is Numa+LP ,can you make it also work as Cluster!
So that we can connect all systems and use all these cores together..

A example i have a i7 5960x 8cores and a i7 960 6cores ..would be 14cores
I think to know that the engine has to adjust to the slowest core..so the systems would work as a 14cores i7 960 then?! if i connect my laptop 4cores..then i would have 18cores at laptop speed..and so on..

Even you have no time for this..do you think it's possible for you to make asmFish Cluster!?

Ipman.

**Comment by Rechenschieber** [comment]
This would be a huge task - you'd have to write new algorithms. So forget about it. I know what I'm talking about - I'm the Rybka cluster guy ;)

**Comment by nabildanial** [comment]
@Rechenschieber Are you Lukas Cimiotti?

**Comment by Ipmanchess** [comment]
It can be a huge task..but nobody believed Mohammed Li that he can do a complete re-write from Stockfish to assembler too..it will fail ..look to it now!
First they told me..don't ask for numa..well it's there..now don't start with Cluster..well we will see ;)
Moh Li is a very talent programmer..how fast he did it ,i have not seen before!

Other programmers tell me every time..they need access to the system and need weeks or month to find out how numa works..well Moh li just sent me a numatest version a few days later he has a working version that runs on this 72cores system @100%! without touching or seeing the system at all!
Same for Peter Osterlund from Texel..two versions and it works!

If i was a programmer and someone say to me,don't start with it..well then i was already busy! ;)

Ipman.

**Comment by Rechenschieber** [comment]
@nabildanial 
Yes, I am Lukas Cimiotti

@Ipmanchess 
using 72 cores/threads is mainly not a question of NJMA, but of processor groups.
[https://msdn.microsoft.com/en-us/library/windows/desktop/dd405503(v=vs.85).aspx](url)

**Comment by Ipmanchess** [comment]
It's because numa-aware that it works..before no engine runs well on this 72cores system..
When the people from a datacenter see me first time testing a few engines..the first and only thing he told me..your software needs Numa..i have direct jump on it..he getting different customers with same problem with their software..just put numa inside..done! and it's about processor groups,nodes..the scaling is just great ,asmFish get around 95% speed up in nodes/sec. when dubbling cores..
It doesn't matter how Moh Li has done it, it works point.

**Comment by hero2017** [comment]
I have to agree with IpMan on both Numa and Cluster aware. While making it cluster aware would indeed be a big job I have no doubt that Mohammed could do it if he was motivated to do so.  Maybe if the SF team decided to take this task seriously then we'd have a SF cluster-aware engine. Wow. I can already see the rich owners of many powerful computers being harnessed together in a cluster and destroying their opponents in tournaments online. The engine's would start overdosing from all that extra ELO!!
